packages/token-bridge-sdk/src/deposits/fetchDeposits.ts

```diff
@@ -0,0 +1,112 @@
+import { Provider } from '@ethersproject/providers'
+import { AssetType } from '../hooks/arbTokenBridge.types'
+import { Transaction } from '../hooks/useTransactions'
+import { utils } from 'ethers'
+import { updateAdditionalDepositData } from '../util/deposits'
+import {
+  fetchDepositsFromSubgraph,
+  FetchDepositsFromSubgraphResult
+} from './fetchDepositsFromSubgraph'
+
+export type DepositETHSubgraphResult = {
+  id: string
+  senderAliased: string
+  destAddr: string
+  value: string
+  msgData: string
+  transactionHash: string
+  blockCreatedAt: string
+}
+
+/* Fetch complete deposits - both ETH and Token deposits from subgraph into one list */
+/* Also fills in any additional data required per transaction for our UI logic to work well */
+/* TODO : Add event logs as well */
+export async function fetchDeposits({
+  address,
+  fromBlock,
+  toBlock,
+  l1Provider,
+  l2Provider,
+  pageSize = 10,
+  pageNumber = 0,
+  searchString = ''
+}: {
+  address: string
+  fromBlock?: number
+  toBlock?: number
+  l1Provider: Provider
+  l2Provider: Provider
+  pageSize?: number
+  pageNumber?: number
+  searchString?: string
+}): Promise<Transaction[]> {
+  const l1ChainId = (await l1Provider.getNetwork()).chainId
```

Comment:
FYI we have a `useChainId` hook which accept a provider. https://github.com/OffchainLabs/arb-token-bridge/blob/master/packages/token-bridge-sdk/src/hooks/useChainId.ts

Reply:
The only problem with hooks is that we cannot use them inside of other functions, just components. Hence I tend to export a lot of hook business logic as an exportable standalone function as well. 
This await will get resolved with time - I am working on optimisation which will take care of unnecessary RPC fetches within this function.

Reply:
Oh, I forgot it wasn't a hook, make sense. Any reason why we're not using hook here (with SWR)?

Reply:
Fetch deposits function is now being used inside a `useDeposits` hook which is SWR cached. This fetchDeposits function is also being unit tested independently.